src/java.base/share/classes/com/sun/crypto/provider/TlsKeyMaterialGenerator.java

```diff
@@ -105,8 +105,8 @@ private SecretKey engineGenerateKey0(byte[] masterSecret) throws GeneralSecurity
 
         SecretKey clientMacKey = null;
         SecretKey serverMacKey = null;
-        SecretKey clientCipherKey = null;
-        SecretKey serverCipherKey = null;
+        SecretKey clientCipherKey;
+        SecretKey serverCipherKey;
```

Comment:
nit: move them down to line 200, e.g. where these two variables are assigned?

Reply:
No. The two variables wouldn't be in scope for the `finally` block on line 276.

Reply:
How about right above the block where they are assigned? The reason that I suggested this is that it's easier to see why no default value needed when they are placed nearer to where they are used.

Reply:
Right above the `try` block? That would work.